app/Services/MentionsService.php

```diff
@@ -652,6 +692,41 @@ protected function prepareEntities(): void
         //dump(count($ids));
         foreach ($entities as $entity) {
             $this->entities[$entity->id] = $entity;
+            $findKey = array_search($entity->id, $ids);
+            unset($ids[$findKey]);
+        }
+        $this->hiddenEntities = $ids;
+    }
+
+    /**
+     * Pre-fetch all private mentioned entities
+     */
+    protected function prepareHiddenEntities(): void
+    {
+        if (!$this->campaign->getNamePrivateEntities()) {
+            return;
+        }
+
+        // Remove those already cached in memory
+        $ids = [];
+        // @phpstan-ignore-next-line
+        foreach ($this->hiddenEntities as $id) {
+            if (!Arr::has($this->privateEntities, $id)) {
+                $ids[] = $id;
+            }
+        }
+
+        // @phpstan-ignore-next-line
+        if (empty($ids)) {
+            return;
+        }
+
+        // Directly get with the mentioned entity types (provided they are valid)
+        // @phpstan-ignore-next-line
+        $entities = Entity::whereIn('id', $ids)->with(['tags:id,name,slug', 'entityType:id,code,is_special'])->withInvisible()->get();
```

Comment:
Shouldn't be needing tags (the ticket only requires the entity type in the span's data properties)